src/test/ui/numeric/numeric-cast-no-fix.stderr

```diff
@@ -2,47 +2,59 @@ error[E0308]: mismatched types
   --> $DIR/numeric-cast-no-fix.rs:10:15
    |
 LL |     x_usize > -1_isize;
-   |               ^^^^^^^^ expected `usize`, found `isize`
+   |     -------   ^^^^^^^^ expected `usize`, found `isize`
+   |     |
+   |     expected because of this binary operation left hand side expression
```

Comment:
Maybe we could say something like "expected because the left-hand side is `$TY`"?

Unrelated, I feel like "binary operation left hand side expression" is a lot of words but not necessarily clearer than just "left-hand side".

Reply:
Changed the wording to be "expected because this is `Ty`", that should be understandable while also short and jargon-free.